Firestore/core/src/firebase/firestore/remote/datastore.mm

```diff
@@ -16,16 +16,85 @@
 
 #include "Firestore/core/src/firebase/firestore/remote/datastore.h"
 
+#include <unordered_set>
+
 #include "Firestore/core/include/firebase/firestore/firestore_errors.h"
+#include "Firestore/core/src/firebase/firestore/core/database_info.h"
+#include "Firestore/core/src/firebase/firestore/remote/grpc_completion.h"
+#include "Firestore/core/src/firebase/firestore/util/executor_libdispatch.h"
 #include "Firestore/core/src/firebase/firestore/util/hard_assert.h"
+#include "absl/memory/memory.h"
+#include "absl/strings/str_cat.h"
 
 namespace firebase {
 namespace firestore {
 namespace remote {
 
-util::Status Datastore::ConvertStatus(grpc::Status from) {
+using core::DatabaseInfo;
+using util::AsyncQueue;
+using util::Status;
+using util::internal::Executor;
+using util::internal::ExecutorLibdispatch;
+
+namespace {
+
+std::unique_ptr<Executor> CreateExecutor() {
+  auto queue = dispatch_queue_create("com.google.firebase.firestore.datastore",
+                                     DISPATCH_QUEUE_SERIAL);
+  return absl::make_unique<ExecutorLibdispatch>(queue);
+}
+
+std::string MakeString(grpc::string_ref grpc_str) {
+  return {grpc_str.begin(), grpc_str.size()};
+}
+
+absl::string_view MakeStringView(grpc::string_ref grpc_str) {
+  return {grpc_str.begin(), grpc_str.size()};
+}
+
+}  // namespace
+
+Datastore::Datastore(const DatabaseInfo &database_info,
+                     AsyncQueue *worker_queue)
+    : grpc_connection_{database_info, worker_queue, &grpc_queue_},
+      dedicated_executor_{CreateExecutor()} {
```

Comment:
nit: "dedicated" is a useful adjective here in that it definitely suggests that the executor is exclusive to Datastore, but doesn't suggest what that purpose is.

Consider matching the name of this executor to the human visible name above. For example, if the above is "com.google.firebase.firestore.rpc" consider naming this `rpc_executor_`.